packages/stash-rs/native/lib.rs

```diff
@@ -29,46 +31,46 @@ type BoxedRecordIndexer = JsBox<RefCell<Indexer>>;
 
 fn init_indexer(mut cx: FunctionContext) -> JsResult<BoxedRecordIndexer> {
     let cbor_schema = cx.argument::<JsBuffer>(0)?;
-
-    let indexer = cx
-        .borrow(&cbor_schema, |buf| {
-            RecordIndexer::decode_from_cbor(buf.as_slice())
-        })
+    let indexer = RecordIndexer::decode_from_cbor(cbor_schema.as_slice(&cx))
         .or_else(|e| cx.throw_error(e.to_string()))?;
-
     Ok(JsBox::new(&mut cx, RefCell::new(Indexer(indexer))))
 }
 
 fn encrypt_record(mut cx: FunctionContext) -> JsResult<JsBuffer> {
     let indexer = cx.argument::<BoxedRecordIndexer>(0)?;
     let cbor_record = cx.argument::<JsBuffer>(1)?;
 
-    let indexer = &mut *indexer.borrow_mut();
+    let indexer = &mut *indexer
+        .try_borrow_mut()
+        .or_else(|_| cx.throw_error("Can't borrow indexer as it is already borrowed"))?;
```

Comment:
What would cause this scenario (not just for the indexer, but for other similar calls as well)?

Reply:
@CDThomas the indexer uses a `RefCell` which does runtime borrow checking. If there was another thread or task or some unsafe code that had previously borrowed this with `.borrow()` or `.borrow_mut()` then this would return `Err`. 

In Neon land the only situation I could think of where this could be an issue is if you have a function that accepts a callback. Something like this:

```
// Rust:
fn encrypt(callback: JsCallback) {
  let indexer = indexer.borrow_mut();
  
  let output = callback();
  
  indexer.encrypt(..);
}

// JavaScript:

encrypt(() => {
  // Calling encrypt would try to borrow the indexer after it's already borrowed
  encrypt(..)
});
```